src/Service.Tests/SqlTests/RestApiTests/RestBodyFlexibleModeTests/MsSqlRestBodyNonStrictModeTests.cs

```diff
@@ -0,0 +1,81 @@
+    /// Class containing integration tests for MsSql- to validate scenarios when we operate in non-strict mode for REST request body,
+    /// i.e. we allow extraneous fields to be present in the request body.
+    /// </summary>
+    [TestClass, TestCategory(TestCategory.MSSQL)]
+    public class MsSqlRestBodyNonStrictModeTests : RestBodyNonStrictModeTests
+    {
+        private static Dictionary<string, string> _queryMap = new()
+        {
+            {
+                "InsertOneWithNonExistingFieldInRequestBody",
+                $"SELECT [categoryid], [pieceid], [categoryName],[piecesAvailable]," +
+                $"[piecesRequired] FROM { _Composite_NonAutoGenPK_TableName } " +
+                $"WHERE [categoryid] = 3 AND [pieceid] = 1 AND [categoryName] = 'SciFi' " +
+                $"AND [piecesAvailable] is NULL AND [piecesRequired] = 1 " +
+                $"FOR JSON PATH, INCLUDE_NULL_VALUES, WITHOUT_ARRAY_WRAPPER"
+            },
+            {
+                "PutOneWithPKFieldsInRequestBody",
+                $"SELECT [categoryid], [pieceid], [categoryName], [piecesAvailable]," +
+                $"[piecesRequired] FROM { _Composite_NonAutoGenPK_TableName } " +
+                $"WHERE [categoryid] = 2 AND [pieceid] = 1 AND [categoryName] = 'SciFi' " +
+                $"AND [piecesAvailable] = 10  AND [piecesRequired] = 5 " +
+                $"FOR JSON PATH, INCLUDE_NULL_VALUES, WITHOUT_ARRAY_WRAPPER"
+            },
+            {
+                "PutOneWithNonExistingFieldInRequestBody",
+                $"SELECT [categoryid], [pieceid], [categoryName], [piecesAvailable]," +
+                $"[piecesRequired] FROM { _Composite_NonAutoGenPK_TableName } " +
+                $"WHERE [categoryid] = 2 AND [pieceid] = 1 AND [categoryName] = 'SciFi' " +
+                $"AND [piecesAvailable] = 10  AND [piecesRequired] = 5 " +
+                $"FOR JSON PATH, INCLUDE_NULL_VALUES, WITHOUT_ARRAY_WRAPPER"
+            },
+            {
+                "PatchOneWithPKFieldsInRequestBody",
+                $"SELECT [categoryid], [pieceid], [categoryName],[piecesAvailable]," +
+                $"[piecesRequired] FROM { _Composite_NonAutoGenPK_TableName } " +
+                $"WHERE [categoryid] = 1 AND [pieceid] = 1 AND [categoryName] = 'SciFi' " +
+                $"AND [piecesAvailable] is NULL AND [piecesRequired] = 0 " +
+                $"FOR JSON PATH, INCLUDE_NULL_VALUES, WITHOUT_ARRAY_WRAPPER"
+            },
+            {
+                "PatchOneWithNonExistingFieldInRequestBody",
+                $"SELECT [categoryid], [pieceid], [categoryName],[piecesAvailable]," +
+                $"[piecesRequired] FROM { _Composite_NonAutoGenPK_TableName } " +
+                $"WHERE [categoryid] = 1 AND [pieceid] = 1 AND [categoryName] = 'SciFi' " +
+                $"AND [piecesAvailable] is NULL AND [piecesRequired] = 0 " +
+                $"FOR JSON PATH, INCLUDE_NULL_VALUES, WITHOUT_ARRAY_WRAPPER"
+            }
```

Comment:
Could you perhaps make a more generic name for this query so you only need one entry in this dictionary? no matter what you change in the request body when:
- `request-body-strict`==false 
- supply non-existing field or include PK in body

The result will be the same given your URI PK values are the same.

Reply:
Since the test comments specify what the test is validating, it would be appropriate to have a generic name like `{Put/Patch/Insert}OneWithExtraneousFieldsInRequestBody`.